docs/migration/README.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "Coming from A2"
```

Comment:
This is directly copied from the "What's new in Apostrophe 3" page. The changes:
- I added the last section with a link to Product Board instead of the separate "Coming soon" page (which was outdated)
- I removed the "Changes in alpha 2" section since we weren't keeping up with adding that for each version.

Please don't judge the PR on the rest of the content.